src/server/scripts/Northrend/Ulduar/HallsOfStone/instance_halls_of_stone.cpp

```diff
@@ -210,12 +212,65 @@ class instance_halls_of_stone : public InstanceMapScript
                 if (GameObject* tribunalDoor = instance->GetGameObject(goTribunalDoorGUID))
                     tribunalDoor->SetGoState(GO_STATE_ACTIVE);
 
+            if (type == BOSS_TRIBUNAL_OF_AGES && data == SPECIAL)
+            {
+                GameObject* pF = instance->GetGameObject(goSkyRoomFloorGUID);
+                pF->SetGoState(GO_STATE_READY);
+            }
+
             if (type == BOSS_TRIBUNAL_OF_AGES && data == DONE)
             {
-                if (GameObject* pA = instance->GetGameObject(goAbedneumGUID))
-                    pA->SetGoState(GO_STATE_ACTIVE);
-                if (GameObject* pF = instance->GetGameObject(goSkyRoomFloorGUID))
-                    pF->SetGoState(GO_STATE_ACTIVE);
+                GameObject* pA = instance->GetGameObject(goAbedneumGUID);
+                GameObject* pK = instance->GetGameObject(goKaddrakGUID);
+                GameObject* pM = instance->GetGameObject(goMarnakGUID);
```

Comment:
These never get null checked

Reply:
additionally would be nice to give the gobs more descriptive names. I know they are initialized with a GUID here so it's kind of clear, but still couldn't hurt to name the like abedneum, kaddrak, marnak, etc.